course2/course_en.md

```diff
@@ -0,0 +1,555 @@
+style: |
+    ol > li {
+        list-style-type:decimal
+    }
+---
+
+# Modern Programming Ideology
+
+## Development Environments & Expressions
+
+### Hongbo Zhang
+
+---
+
+# MoonBit
+
+- A modern, statically-typed, **multi-paradigm** programming language
+with newbie-friendly lightweight syntax.
+- Website: [moonbitlang.com](https://www.moonbitlang.com/)
+
+---
+
+# Development Environments
+
+---
+
+# MoonBit Development Environments
+
+- **Tools**
+    - **VS Code extensions**: language server, package-level build tool...
+    - **Command-line tools**: project-level build, project testing, dependencies management...
+- **Environments**
+    - **Browser-based** (without command-line tools)
+    - **Cloud-native**:
+        - [Coding](https://coding.net/)
+        - [Gitpod](https://www.gitpod.io/)
+        - [GitHub.dev](https://github.dev/)
+        - ...
+    - **Local**
+
+---
+
+# Browser-Based Environment
+
+- Visit [try.moonbitlang.com](https://try.moonbitlang.com), or click the "Try" tab on our [Website](https://moonbitlang.com).
+- The environment offers the following features:
+    - File creation and execution
+    - Code sharing
+- Sample programs are provided for beginners to learn MoonBit.
+
+---
+
+# Cloud-Native Environment (Coding)
+
+- On-demand development environments based on remote servers
+- Provided by cloud-native development environment providers, e.g., Coding.
+- Create or clone a repo, launch the environment, and then install the "MoonBit Language" extension.
+- (Advanced) Install the [command-line tools](https://www.moonbitlang.com/download/) or clone the [cloud-native development template](https://github.com/peter-jerry-ye/moonbit-template), and refer to the [_MoonBit's Build System Tutorial_](https://www.moonbitlang.com/docs/build-system-tutorial/) for further guidance.
+
+---
+
+# Local Environment
+
+- Install [VS Code](https://code.visualstudio.com/) or [VS Codium](https://vscodium.com/), and then install the "MoonBit Language" extension.
+- (Advanced) Install the [command-line tools](https://www.moonbitlang.com/download/) (cross-platform) and refer to the [_MoonBit's Build System Tutorial_](https://www.moonbitlang.com/docs/build-system-tutorial/) for further guidance.
+
+---
+
+# MoonBit Expressions
+
+---
+
+# A Typical MoonBit Program
+
+```moonbit
+// top-level function definition
+fn num_water_bottles(num_bottles: Int, num_exchange: Int) -> Int {
+  // local function definition
+  fn consume(num_bottles, num_drunk) {
+    // conditional expression
+    if num_bottles >= num_exchange {
+      // variable binding
+      let num_bottles = num_bottles - num_exchange + 1
+      let num_drunk = num_drunk + num_exchange
+      // function application
+      consume(num_bottles, num_drunk)
+    } else {
+      num_bottles + num_drunk
+    }
+  }
+  consume(num_bottles, 0)
+}
+
+// entry point
+fn init {
+  // statements
+  assert(num_water_bottles(9, 3) == 13)
+  assert(num_water_bottles(15, 4) == 19)
+}
```

Comment:
I did not modify the code, just translated the comments. However, the program needs to be updated to comply with the latest version of MoonBit.

Reply:
Yes, you can update it. I updated `lec2_script.md` and `lec2.md`, maybe you can check that

Reply:
> Yes, you can update it. I updated `lec2_script.md` and `lec2.md`, maybe you can check that

I am not sure whether it is appropriate to introduce `test` as the entry point since we would rather introduce `init` as the entry point.

Reply:
We will introduce `main` (instead of `init`) later. For now, introducing `test` is fine.